src/med_bench/mediation.py

```diff
@@ -362,64 +250,15 @@ def mediation_g_formula(y, t, m, x, interaction=False, forest=False,
                    1e-5 and 1e5
     """
-    if regularization:
-        alphas = ALPHAS
-        cs = ALPHAS
-    else:
-        alphas = [TINY]
-        cs = [np.inf]
-    n = len(y)
-    if len(x.shape) == 1:
-        x = x.reshape(-1, 1)
-    if len(m.shape) == 1:
-        mr = m.reshape(-1, 1)
-    else:
-        mr = np.copy(m)
-    if len(t.shape) == 1:
-        t = t.reshape(-1, 1)
-    t0 = np.zeros((n, 1))
-    t1 = np.ones((n, 1))
-    m0 = np.zeros((n, 1))
-    m1 = np.ones((n, 1))
 
-    if crossfit < 2:
-        train_test_list = [[np.arange(n), np.arange(n)]]
-    else:
-        kf = KFold(n_splits=crossfit)
-        train_test_list = list()
-        for train_index, test_index in kf.split(x):
-            train_test_list.append([train_index, test_index])
-    mu_11x, mu_10x, mu_01x, mu_00x, f_00x, f_01x, f_10x, f_11x = \
-        [np.zeros(n) for h in range(8)]
-
-    for train_index, test_index in train_test_list:
-        if not forest:
-            y_reg = RidgeCV(alphas=alphas, cv=CV_FOLDS)\
-                .fit(get_interactions(interaction, x, t, mr)[train_index, :], y[train_index])
-            pre_m_prob = LogisticRegressionCV(Cs=cs, cv=CV_FOLDS)\
-                .fit(get_interactions(interaction, t, x)[train_index, :], m.ravel()[train_index])
-        else:
-            y_reg = RandomForestRegressor(n_estimators=100, min_samples_leaf=10)\
-                .fit(get_interactions(interaction, x, t, mr)[train_index, :], y[train_index])
-            pre_m_prob = RandomForestClassifier(n_estimators=100, min_samples_leaf=10)\
-                .fit(get_interactions(interaction, t, x)[train_index, :], m.ravel()[train_index])
-        if calibration:
-            m_prob = CalibratedClassifierCV(pre_m_prob, method=calib_method)\
-                .fit(get_interactions(
-                    interaction, t, x)[train_index, :], m.ravel()[train_index])
-        else:
-            m_prob = pre_m_prob
-        mu_11x[test_index] = y_reg.predict(get_interactions(interaction, x, t1, m1)[test_index, :])
-        mu_10x[test_index] = y_reg.predict(get_interactions(interaction, x, t1, m0)[test_index, :])
-        mu_01x[test_index] = y_reg.predict(get_interactions(interaction, x, t0, m1)[test_index, :])
-        mu_00x[test_index] = y_reg.predict(get_interactions(interaction, x, t0, m0)[test_index, :])
-        f_00x[test_index] = m_prob.predict_proba(get_interactions(interaction, t0, x)[test_index, :])[:, 0]
-        f_01x[test_index] = m_prob.predict_proba(get_interactions(interaction, t0, x)[test_index, :])[:, 1]
-        f_10x[test_index] = m_prob.predict_proba(get_interactions(interaction, t1, x)[test_index, :])[:, 0]
-        f_11x[test_index] = m_prob.predict_proba(get_interactions(interaction, t1, x)[test_index, :])[:, 1]
+    classifier_y, classifier_m = _get_y_m_classifiers(regularization, forest, calibration, calib_method)
+    f, mu = _estimate_f_mu(t, m, x, y, crossfit, classifier_y, classifier_m, interaction)
```

Comment:
naming again, regressor and not classifier for y

Reply:
This has been addressed in 3f1b34577643b37b886202ab5873e7b5a9054a40